modin/pandas/groupby.py

```diff
@@ -255,15 +255,21 @@ def _shift(data, periods, freq, axis, fill_value, is_set_nan_rows=True):
             result = _shift(
                 self._df, periods, freq, axis, fill_value, is_set_nan_rows=False
             )
-            new_idx_lvl_arrays = np.concatenate(
-                [self._df[self._by.columns].values.T, [list(result.index)]]
-            )
-            result.index = pandas.MultiIndex.from_arrays(
-                new_idx_lvl_arrays,
-                names=[col_name for col_name in self._by.columns]
-                + [result._query_compiler.get_index_name()],
-            )
-            result = result.dropna(subset=self._by.columns).sort_index()
+            if PandasCompatVersion.CURRENT != PandasCompatVersion.LATEST:
```

Comment:
Should this stuff be in `Python36CompatibleDataFrameGroupBy.shift`?

Reply:
ideally yes, but that would require certain amount of logic duplication, as I cannot sort on index without building the index first, I think this way it's easier to go forward, though less clean that desired

Reply:
https://github.com/modin-project/modin/issues/5086